### ISSUE TITLE
NamingStrategy messageType

### ISSUE THREAD

**Issue by duncanbrown** [justeat/JustSaying#358]
Consumers can customise topic/queue names on a per-message-type basis by providing an `INamingStrategy`. However, this is somewhat difficult to work with because the message type itself is not passed into the relevant methods, but rather a stringified representation (coming from https://github.com/justeat/JustSaying/blob/4050749a2b3cef9cdf566a21ba513ea5241c14ef/JustSaying/Extensions/TypeExtensions.cs#L12). Getting back from this string to the original type isn't trivial - especially in the case of generic types - which is problematic for consumers who want to base the aws resource name on the type in question.

I guess the simplest fix would be to add an additional `Type` parameter to the `INamingStrategy` methods, for the message's type. This would be a breaking change, but I notice there's a major version release in the pipeline anyway: would you be open to a pr along these lines?

**Comment by AnthonySteele** [comment]
Seems reasonable.  Is there a PR for this?

You might even change the signature to `string GetTopicName(string topicName, Type messageType);` and the naming strategy could call `ToTopicName` on that type _if_ that was the desired behaviour.

**Comment by duncanbrown** [comment]
No PR yet, but happy to put one together @AnthonySteele. On your suggestion of dropping the string and just using the type: that would suit my purposes, but I notice that the`ToTopicName` extension is currently in an internal class - would you be happy for that to be publically exposed, or is it considered a private implementation detail?

**Comment by AnthonySteele** [comment]
You're right, it is internal. I would be happy to see that as a `public static` method. Extension or not is a matter of style, it doesn't make a big difference.
